### ISSUE TITLE
[Deviantart] It´s possible to speed up the process of download?

### ISSUE THREAD

**Issue by joselpg** [4pr0n/ripme#584]
I think the process is too low when we have a lot of images. In my case it starts downloading 24 per page, and delay 3 minutes for start with the next page. So just imagine download a big collection

**Comment by Hrxn** [comment]
Download speed is always limited by the server, so I don't think there is much that can be done here..

**Comment by rautamiekka** [comment]
@Hrxn is right, the speed you get with DA is **very** variable. You can try 2 threads (I wouldn't try more than that, but that's me) since it's a little faster even if DA's end is busy; I've done many folders which have had hundreds of items each, so I know from experience.

If you drop the timeout to 12000 milliseconds, meaning 12 seconds, you might get some boost, at least I seem to have, versus 60000 (60s) or even bigger, where it took even 15 minutes between pages. Seemingly that timeout is how long RipMe waits for reply and re-tries when nothing happens, which'd explain why it takes so long.

**_The bottom line:_** don't try too many threads and too low timeouts cuz although DA still hasn't IP-banned me for ripping tons of stuff, they could get wise anytime and suddenly put a range of rippers behind a firewall, which could end uglily. 2 threads and 12s sound like the reasonable max to me to keep using.

**Comment by joselpg** [comment]
@Hrxn You´re right. Sound logical for me.
@rautamiekka Yes. Is configured by default with: 50 Maximum threads and 60000 milliseconds of timeout. I didn't use your recommended configuration. But I tried with a similar,  and there isn´t a significant diference for speed up the velocity. I´m donwloading with 24 threads per page, I think is the max. capacity for DA use. Because only 24 images are hosted per page. And about timeout. I´m not sure, but I used 10000 ms.
In shot, There´s not much you can do about the spped up. I have 20Mbps, and according to my calculations. It will take me for over 30 hours to download 17000 images of my collection.

**Comment by rautamiekka** [comment]
^ 24 threads is just asking for trouble, from both DA and your connection, and if you've 20Mbps down you're not giving it much per thread.

**Comment by metaprime** [comment]
Looks like this issue is addressed as "by design".
Closing as client-side / by-design / wontfix.
